docs/src/config.md

```diff
@@ -0,0 +1,87 @@
+# [Rule configurations and calling back into AD](@id config)
+
+[`RuleConfig`](@ref) a method for making rules conditionally defined based on the presence of certain features in the AD system.
+Once key such feature is the ability to perform AD either in forwards or reverse mode or both.
+
+
+## Writing rules that call back into AD
+
+To define e.g. rules for higher order functions, it is useful to be able to call back into the AD system to get it to do some work for you.
+
+For example the rule for reverse mode AD for map might like to use forward mode AD if one is available.
+Particularly for the case where only a single input collection is being mapped over.
```

Comment:
Might be cleanest to be really concrete and say a `Vector{Float64}` or something, rather than single input collection, because you don't want to use forwards-mode for lots of unary functions (if their inputs are high-dimensional)

Reply:
I do so in the code.
More importantly, AFAIK it isn't even possible to do this for other tyles, as we have no way to get a basis to propagate.
But for `Real`s the basis just so happens to be the multiplicative identity.